crypto/s2n_certificate.h

```diff
@@ -40,11 +42,13 @@ struct s2n_cert_chain_and_key {
     s2n_cert_private_key *private_key;
     struct s2n_blob ocsp_status;
     struct s2n_blob sct_list;
-    char server_name[S2N_MAX_SERVER_NAME];
+    GENERAL_NAMES *san_names;
```

Comment:
I was thinking about wrapping this in something like `s2n_dns_names` or something similar so we can make it easy to change the implementation. 

We can decide to wrap it in a different PR.